jobmanager/Job.h

```diff
@@ -48,4 +48,38 @@ class CountedJob : public Job
     utils::ScopedIncrement _jobsCounterIncrement;
 };
 
+template <class Callable>
+class CallableJob final : public Job
+{
+public:
+    explicit CallableJob(const Callable& callable) : _callable(callable) {}
+    explicit CallableJob(Callable&& callable) : _callable(std::move(callable)) {}
+
+    void run() final { _callable(); }
+
+private:
+    Callable _callable;
+};
+
+template <class Callable>
```

Comment:
Generic callable. The only restriction is that needs to implement `operator()()`. This is has some advantages over give a `std::function.`

1. We can use the direct type returned by `utils::bind` which creates one less level of indirection as the result is a `final` type and does not need to invoke more virtual methods
2. We can use lambda functions directly. The lambda function itself does not created dynamic memory (dynamic memory can happens when we transform the lambda on std::function).

Reply:
can we put this comment above the CallableJob?